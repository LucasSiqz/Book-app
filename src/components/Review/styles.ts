import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 33.5rem;
`

export const Image = styled.img`
  width: 100%;
  height: 18.1rem;
`

export const Info = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    padding: ${theme.spacings.xsmall} ${theme.spacings.xxsmall};
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    margin-bottom: 0.6rem;
  `}
`

export const SubInfo = styled.div`
  display: flex;
  align-items: center;

  p {
    font-size: 0.8rem;
    color: #6a6767;
  }
`

export const Divider = styled.div`
  ${({ theme }) => css`
    width: 0.2rem;
    height: 0.2rem;
    border-radius: 50%;
    background: #6a6767;
    margin-top: 0.2rem;
    margin-left: ${theme.spacings.xsmall};
    margin-right: ${theme.spacings.xxsmall};
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    color: #6b6b6b;
    font-size: ${theme.font.sizes.xsmall};
    margin-top: 0.5rem;
  `}
`
