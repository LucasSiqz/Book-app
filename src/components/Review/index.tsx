import * as S from './styles'

export type ReviewProps = {
  title: string
  author: string
  description: string
  viewsCount: number
  weeksAgo: number
}

const Review = ({
  title,
  author,
  description,
  viewsCount,
  weeksAgo
}: ReviewProps) => (
  <S.Wrapper>
    <S.Image src="img/Review.png" alt={`${title}`} />

    <S.Info>
      <S.Title>{title}</S.Title>
      <S.SubInfo>
        <p>{author}</p>
        <S.Divider />
        <p>{`${viewsCount} Views`}</p>
        <S.Divider />
        <p>{`${weeksAgo} Week(s) ago`}</p>
      </S.SubInfo>

      <S.Description>{description}</S.Description>
    </S.Info>
  </S.Wrapper>
)

export default Review
