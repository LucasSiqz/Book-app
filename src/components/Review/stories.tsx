import { Story, Meta } from '@storybook/react/types-6-0'

import Review, { ReviewProps } from '.'

export default {
  title: 'Review',
  component: Review,
  args: {
    title: "Don't Make Me Think",
    author: 'Jesse Showalter',
    description:
      '"Dont Make Me Think" by Steve Krug is one of the first books I read when I was getting into digital design. It helped me move from designing things that just "look nice" to designing things that are usable, useful, memorable and simple to learn and use. ',
    viewsCount: 5.2,
    weeksAgo: 1
  }
} as Meta

export const Default: Story<ReviewProps> = (args) => <Review {...args} />
