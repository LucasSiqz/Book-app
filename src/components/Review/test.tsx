import { render, screen } from 'utils/test-utils'

import Review from '.'

const props = {
  title: "Don't Make Me Think",
  author: 'Jesse Showalter',
  image: 'img/Review.png',
  description: 'description',
  viewsCount: 5.2,
  weeksAgo: 1
}

describe('<Review >', () => {
  it('should render correctly', async () => {
    const { container } = render(<Review {...props} />)

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      props.image
    )
    expect(
      screen.getByRole('heading', { name: /Don't Make Me Think/i })
    ).toBeInTheDocument()
    expect(await screen.findByText(/Jesse Showalter/i)).toBeInTheDocument()
    expect(await screen.findByText(/5.2 Views/i)).toBeInTheDocument()
    expect(await screen.findByText('1 Week(s) ago')).toBeInTheDocument()
    expect(await screen.findByText(/description/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
