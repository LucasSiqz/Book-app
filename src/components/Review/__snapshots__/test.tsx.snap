// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review > should render correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 33.5rem;
}

.c1 {
  width: 100%;
  height: 18.1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1.6rem 0.8rem;
}

.c3 {
  font-size: 1.4rem;
  margin-bottom: 0.6rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 p {
  font-size: 1rem;
  color: #6a6767;
}

.c5 {
  width: 0.2rem;
  height: 0.2rem;
  border-radius: 50%;
  background: #6a6767;
  margin-top: 0.2rem;
  margin-left: 1.6rem;
  margin-right: 0.8rem;
}

.c6 {
  color: #6b6b6b;
  font-size: 1.2rem;
  margin-top: 0.5rem;
}

<div
  class="c0"
>
  <img
    alt="Don't Make Me Think"
    class="c1"
    src="img/Review.png"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      Don't Make Me Think
    </h2>
    <div
      class="c4"
    >
      <p>
        Jesse Showalter
      </p>
      <div
        class="c5"
      />
      <p>
        5.2 Views
      </p>
      <div
        class="c5"
      />
      <p>
        1 Week(s) ago
      </p>
    </div>
    <p
      class="c6"
    >
      description
    </p>
  </div>
</div>
`
